is/object_of.ts

```diff
@@ -43,22 +43,24 @@ export function isObjectOf<
     ...Object.keys(predObj),
     ...Object.getOwnPropertySymbols(predObj),
   ].map((k) => [k, predObj[k]]);
-  return annotate(
-    rewriteName(
-      (x: unknown): x is ObjectOf<T> => {
-        if (
-          x == null ||
-          typeof x !== "object" && typeof x !== "function" ||
-          Array.isArray(x)
-        ) return false;
-        return preds.every(([k, pred]) => pred((x as T)[k]));
-      },
-      "isObjectOf",
-      predObj,
-    ),
-    "predObj",
+  const pred = rewriteName(
+    (x): x is ObjectOf<T> => {
+      if (!isRecordT<T>(x)) return false;
+      return preds.every(([k, pred]) => pred(x[k]));
+    },
+    "isObjectOf",
     predObj,
   );
+  return annotate(pred, "predObj", predObj);
+}
+
+function isRecordT<T extends Record<PropertyKey, Predicate<unknown>>>(
+  x: unknown,
+): x is T {
```

Comment:
Is not property values ​​of `x` be `unknown` instead of `Predicate<unknown>`?
```suggestion
function isRecordT(x: unknown): x is Record<PropertyKey, unknown> {
```

Reply:
You're right. Thanks